keras/engine/data_adapter.py

```diff
@@ -1172,6 +1172,10 @@ def __init__(self,
   def _configure_dataset_and_inferred_steps(self, strategy, x, steps_per_epoch,
                                             class_weight, distribute):
     """Configure the `_dataset` and `_inferred_steps` attributes."""
+    if tf.size(x)==0:
```

Comment:
Note that this might not be correct when x is a dataset or iterator. Also please add space before and after "=="